### ISSUE TITLE
Loading exported ONNX IR 10 model in Rust example

### ISSUE THREAD

**Issue by Jobarion** [ultralytics/ultralytics#16504]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Integrations, Other

### Bug

Exporting a recently custom trained model to onnx, and trying to load it in the [Rust example](https://github.com/ultralytics/ultralytics/tree/main/examples/YOLOv8-ONNXRuntime-Rust) fails with the following error:
```
Error: Failed to create ONNX Runtime session: Load model from .\best.onnx failed:C:\a\_work\1\s\onnxruntime\core\graph\model.cc:150 onnxruntime::Model::Model Unsupported model IR version: 10, max supported IR version: 9
```

The model was exported using `yolo export model=best.pt format=onnx simplify`.

### Environment

Ultralytics YOLOv8.2.71 🚀 Python-3.12.4 torch-2.3.1+cu121 CUDA:0 (NVIDIA GeForce RTX 2070, 8192MiB)
Setup complete ✅ (12 CPUs, 31.9 GB RAM, 1323.1/1862.4 GB disk)

OS                  Windows-10-10.0.19045-SP0
Environment         Windows
Python              3.12.4
Install             pip
RAM                 31.90 GB
CPU                 AMD Ryzen 5 3600X 6-Core Processor
CUDA                12.1

numpy               ✅ 1.26.3<2.0.0,>=1.23.0
matplotlib          ✅ 3.9.1>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 10.2.0>=7.1.2
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.14.0>=1.4.1
torch               ✅ 2.3.1+cu121>=1.8.0
torchvision         ✅ 0.18.1+cu121>=0.9.0
tqdm                ✅ 4.66.4>=4.64.0
psutil              ✅ 6.0.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.2>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.0>=2.0.0

### Minimal Reproducible Example

Export model using `yolo export model=best.pt format=onnx simplify`.
Run `cargo run --release -- --model best.onnx --source test.jpg  --trt --fp16 --plot`

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Jobarion, thank you for your interest in Ultralytics 🚀! We appreciate your report and are here to help.

If this is a 🐛 Bug Report, please ensure you provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to assist us in debugging it. From your description, it seems that the issue might relate to model version compatibility with ONNX Runtime in Rust.

For initial guidance, please make sure you've followed all the setup instructions provided in the [Rust example](https://github.com/ultralytics/ultralytics/tree/main/examples/YOLOv8-ONNXRuntime-Rust) and check if your ONNX Runtime supports the required IR version.

Join the Ultralytics community for more dynamic support. For real-time chat, head to [Discord](https://ultralytics.com/discord) 🎧. Prefer in-depth discussions? Visit our [Discourse](https://community.ultralytics.com), or check out our [Subreddit](https://reddit.com/r/ultralytics) for broader engagement.

## Upgrade

Ensure your setup is up-to-date with the latest `ultralytics` package and dependencies. Use a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLOv8 is supported in diverse environments (with [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn) preinstalled):

- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Cloud**: [Google Cloud](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/), [AWS](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/), [Docker](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/)

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [CI tests](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) verify that YOLOv8 functions correctly across platforms.

An Ultralytics engineer will reach out soon to assist further. Thank you for bringing this to our attention! 😊

**Comment by vodan37** [comment]
Same problem with triton.
